plugin/better-whitespace.vim

```diff
@@ -234,60 +284,92 @@ function! s:GotoTrailingWhitespace(search_backwards, from, to)
     let @/=_s
 endfunction
 
-" Strip whitespace on file save
+
+" Sets up (or removes) all auto commands in the buffer, after checking the
+" per-buffer settings. Also performs an initial highlighting (or clears it).
+function! <SID>SetupAutoCommands()
+    augroup better_whitespace
+        " Reset all auto commands in group
+        autocmd!
+
+        if <SID>ShouldHighlight()
+            if s:better_whitespace_initialized == 0
+                call <SID>WhitespaceInit()
+            endif
+
+            " Highlight extraneous whitespace at the end of lines, but not the current line in insert mode.
+            call <SID>HighlightEOLWhitespace()
+            autocmd CursorMovedI,InsertEnter * call <SID>HighlightEOLWhitespaceExceptCurrentLine()
+            autocmd InsertLeave,BufReadPost * call <SID>HighlightEOLWhitespace()
+
+            if g:current_line_whitespace_disabled_soft == 0
+                " Using syntax: clear whitespace highlighting when leaving buffer
+                autocmd BufWinLeave * call <SID>ClearHighlighting()
+
+                " Do not highlight whitespace on current line in insert mode
+                autocmd CursorMovedI * call <SID>HighlightEOLWhitespaceExceptCurrentLine()
+
+                " Do not highlight whitespace on current line in normal mode?
+                if g:current_line_whitespace_disabled_hard == 1
+                    autocmd CursorMoved * call <SID>HighlightEOLWhitespaceExceptCurrentLine()
+                endif
+            endif
+
+        elseif s:better_whitespace_initialized == 1
+            " Clear highlighting if it disabled, as it might have just been toggled
+            call <SID>ClearHighlighting()
+        endif
+
+        " Strip whitespace on save if enabled.
+        if <SID>ShouldStripWhitespace()
+            autocmd BufWritePre * call <SID>StripWhitespaceOnSave(v:cmdbang)
+        endif
+
+    augroup END
+endfunction
+
+" Check & setup auto commands upon enter & load, and again on filetype change.
+autocmd FileType,WinEnter,BufWinEnter * call <SID>SetupAutoCommands()
+autocmd ColorScheme * call <SID>WhitespaceInit()
```

Comment:
Whitespace-only lines are not highlighted after colorscheme changes with this. I think we also need to call `HighlightEOLWhitespace()` to ensure everything gets re-highlighted that should be.

Reply:
`HighlightEOLWhitespace()` gets called inside `SetupAutoCommands()` if the `ShouldHighlight()` is true. Whitespace-only lines work fine with me with colorscheme changes. What is the value of your `g:better_whitespace_skip_empty_lines` ?

Reply:
Ah, never mind. It looks like it's actually [vim-indent-guides](https://github.com/nathanaelkane/vim-indent-guides) that's stomping these for me, since it's also listening for `ColorScheme` 😕